plugins/inputs/gnmi/gnmi.go

```diff
@@ -385,19 +461,31 @@ func (c *GNMI) handleSubscribeResponseUpdate(address string, response *gnmiLib.S
 }
 
 // HandleTelemetryField and add it to a measurement
-func (c *GNMI) handleTelemetryField(update *gnmiLib.Update, tags map[string]string, prefix string) (string, map[string]interface{}) {
-	gpath, aliasPath, err := c.handlePath(update.Path, tags, prefix)
+func (c *GNMI) handleTelemetryField(update *gnmiLib.Update, tags map[string]string, prefix string, updateTags bool) (string, map[string]interface{}, bool) {
+	// Tracks if there are tags sent in the update.
+	// If not, then the update is a simple field and
+	// we must search embedded tags in subsequent updates
+	lacksTagsPerUpdate := false
+	lengthTagsBeforeUpdate := len(tags)
+
+	gPath, aliasPath, err := c.handlePath(update.Path, tags, prefix, updateTags)
 	if err != nil {
 		c.Log.Errorf("handling path %q failed: %v", update.Path, err)
 	}
 
+	lengthTagsAfterUpdate := len(tags)
+
+	if lengthTagsBeforeUpdate == lengthTagsAfterUpdate {
+		lacksTagsPerUpdate = true
+	}
```

Comment:
How about
```suggestion
	lengthTagsBeforeUpdate := len(tags)

	gPath, aliasPath, err := c.handlePath(update.Path, tags, prefix, updateTags)
	if err != nil {
		c.Log.Errorf("handling path %q failed: %v", update.Path, err)
		
		// WHAT SHOULD WE DO HERE?
	}

	lengthTagsAfterUpdate := len(tags)
	lacksTagsPerUpdate := lengthTagsBeforeUpdate == lengthTagsAfterUpdate
```

I wonder why you don't return in the error path... Is there a specific reason to continue processing?